File: src/main/java/pl/tsimur/jpatest/mapper/UserMapper.java
package pl.tsimur.jpatest.mapper;

import org.mapstruct.Mapper;
import pl.tsimur.jpatest.model.User;
import pl.tsimur.jpatest.model.dto.LoginDto;
import pl.tsimur.jpatest.model.dto.UserMiniDto;

import java.util.List;

@Mapper
public interface UserMapper {

    LoginDto toLoginDto(User user);

    User fromLoginDto(LoginDto user);

    List<UserMiniDto> toUserMiniDto(List<User> users);

    UserMiniDto toUserMiniDto(User user);


}

File: src/main/kotlin/pl/timur/jpatest/mapper/UserMapper.kt
package pl.timur.jpatest.mapper

import org.mapstruct.Mapper
import pl.timur.jpatest.model.User
import pl.timur.jpatest.model.dto.LoginDto
import pl.timur.jpatest.model.dto.UserMiniDto

@Mapper
interface UserMapper {
    fun toLoginDto(user: User): LoginDto
    fun fromLoginDto(user: LoginDto): User
    fun toUserMiniDto(users: List<User>): List<UserMiniDto>
    fun toUserMiniDto(user: User): UserMiniDto
}